source/s3_meta_request.c

```diff
@@ -22,8 +25,24 @@ struct s3_meta_request_binding {
     /* Reference proxy to python self */
     PyObject *self_py;
 
+    /**
+     * FILE pointer. If set, it handles file operation from C land to reduce the cost
+     * passing chunks from C into python. One for recv/writing, the other for send/reading
+     **/
+    FILE *recv_file;
+    FILE *send_file;
```

Comment:
why do we need `send_file`? doesn't `aws_input_stream_new_from_file()` already handle opening and closing a `FILE*`?
if you to open or close it in some special way, use `aws_input_stream_new_from_open_file()` instead